This PR include ecf directory.

lgannoaa opened the pull request:
As directed by Rahul Mahajan this ecflow PR will include "PBS -l place" from the Google sheet column L (snapshot at 5:00p January 5, 2022) and "PBS -l walltime" from current operation.  This PR will NOT be tested at time of submit.

The snapshot URL https://docs.google.com/spreadsheets/d/1sef7gAq5OZAP_ioySKKwcMmmOkhlM0RMO67XTIoMlDg/edit?usp=sharing

gldas is updated to gldas.fd (e757e6ed)
  PBS -l for job gdas gldas has been modified to:
    #PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128

This PR is only for update under the $HOMEgfs/ecf location for wcoss2 GFS transition.

lgannoaa committed:
As directed by Rahul Mahajan this ecflow PR will include "PBS -l place" from the Google sheet column L (snapshot at 5:00p January 5, 2022) and "PBS -l walltime" from current operation.  This PR will NOT be tested at time of submit.

The snapshot URL https://docs.google.com/spreadsheets/d/1sef7gAq5OZAP_ioySKKwcMmmOkhlM0RMO67XTIoMlDg/edit?usp=sharing

gldas is updated to gldas.fd (e757e6ed)
  PBS -l for job gdas gldas has been modified to:
    #PBS -l place=vscatter,select=1:mpiprocs=128:ompthreads=1:ncpus=128

This PR is only for update under the $HOMEgfs/ecf location for wcoss2 GFS transition.

aerorahul commented on ecf/include/head.h:
```diff
@@ -4,45 +4,83 @@ export PS4='+ $SECONDS + '
+export ecflow_ver=%ecflow_ver%
+
+if [ -d /apps/ops/prod ]; then # On WCOSS2
+####  set +x
```
please retain this `set +x` and `set -x` on line 27.

aerorahul commented on ecf/include/head.h:
```diff
@@ -4,45 +4,83 @@ export PS4='+ $SECONDS + '
+  module load intel/19.1.3.304
+  module load cray-mpich/8.1.9
+  module load cray-pals/1.0.17
+  module load python/3.8.6
```
Curious, why `python` is required in `head.h`?

aerorahul replied:
Also.  Does every job need the mpich module?
By that extension, we could pretty much emulate `load_fv3gfs_modules.sh` in `head.h` and not have to load any module in any of the `ecf` scripts.

aerorahul commented on ecf/include/head.h:
```diff
@@ -4,45 +4,83 @@ export PS4='+ $SECONDS + '
-ECF_TRYNO=${ECF_TRYNO}
-ECF_RID=${ECF_RID}
-ENDFILE
+export HOMEgfs=/lfs/h2/emc/global/noscrub/Lin.Gan/git/feature-ops-wcoss2
```
Shouldn't this have a `Developer Overwrite` wrap around it?

aerorahul commented on ecf/scripts/gdas/atmos/analysis/jgdas_atmos_analysis.ecf:
```diff
@@ -29,14 +21,14 @@ export model=gfs
-module load python/${python_ver}
-module load NetCDF-parallel/${netcdf_parallel_ver}
-module load HDF5-parallel/${hdf5_parallel_ver}
+module load netcdf/${netcdf_ver}
+module load hdf5/${hdf5_ver}
```
This order needs to be reversed.

aerorahul replied:
This comment is for everywhere the netcdf and hdf5 modules are loaded.
netcdf depends on hdf5 and the latter needs to be loaded before.

It is likely there are other module loads that are not following their dependency tree.
Reviewing that will take a long time at this point.

aerorahul commented on ecf/scripts/gdas/atmos/analysis/jgdas_atmos_analysis.ecf:
```diff
@@ -29,14 +21,14 @@ export model=gfs
 module load crtm/${crtm_ver}
+module load libjpeg/${libjpeg_ver}
 module load grib_util/${grib_util_ver}
+module load wgrib2/${wgrib2_ver}
```
Where is `wgrib2` used in the analysis job?

aerorahul commented on ecf/scripts/gdas/atmos/analysis/jgdas_atmos_analysis.ecf:
```diff
@@ -29,14 +21,14 @@ export model=gfs
-module load python/${python_ver}
-module load NetCDF-parallel/${netcdf_parallel_ver}
-module load HDF5-parallel/${hdf5_parallel_ver}
+module load netcdf/${netcdf_ver}
```
Where are the PrgIntel, intel, mpich, cray-pals etc modules loaded?
I thought they are needed to execute.

I see in jgdas_atmos_dump.ecf, some of these are loaded.
So I wonder why the inconsistency?

aerorahul reviewed (commented):
It is going to be difficult to review this in 1 sitting.
So here are my first comments looking at the first few files.

KateFriedman-NOAA commented on ecf/scripts/gdas/atmos/obsproc/dump/jgdas_atmos_tropcy_qc_reloc.ecf:
```diff
@@ -1,16 +1,13 @@
-#BSUB -R span[ptile=1]
-#BSUB -cwd /tmp
+#PBS -S /bin/bash
+#PBS -N %E%gdas_atmos_tropcy_qc_reloc_%CYC%
```
We are supposed to remove the `%E%` text from the jobname line:
```
#PBS -N gdas_atmos_tropcy_qc_reloc_%CYC%
```

See the request to remove `%E%` in the NCO checklist for hand-off:
https://docs.google.com/document/d/13p_wvsKTejk7M5q2KMyOeijH4shghlJrTpQlZkThtDI/edit#

Please remove `%E%` from all ecf scripts, thanks!

KateFriedman-NOAA commented:
As mentioned in another comment above...we are supposed to remove the `%E%` text from the jobname line. We are also supposed to make sure that "job name (-N) start with package name in PBSpro submission". See both requirements from NCO in this checklist for hand-off:
https://docs.google.com/document/d/13p_wvsKTejk7M5q2KMyOeijH4shghlJrTpQlZkThtDI/edit#

Please address the needed changes and other questions I tagged you on in that checklist document, thanks!

WalterKolczynski-NOAA commented:
Please remember that the PR descriptions become the commit message. Statements like "As directed by..." do not really belong in the git log (and definitely not in the commit title). If necessary they can be added as a comment on the PR. Please use good commit message practices when writing commit messages (see the wiki).

lgannoaa commented:
Here is a reply for all question above:
1. %E% has been removed.
2. analysis jobs do not use wgrib2. It removed and under testing.
3. Module position is not been changed in over two weeks. The output has been reviewed by code managers during the 20210824 testing run. Prefer not to change at this time without testing.
4. NCO does not use the ecf/include directory. They have their production version that handle all models.
5. NCO require all job script in ecflow load its own model. It is not allowed to load all modules on every jobs.

lgannoaa committed:
Incremental PR with small changes to remove E from each ecflow script file.
GLDAS job to use the updated resource and walltime in the google sheet.
HDF error casued GDAS ENKF post jobs to fail during testing.

aerorahul commented on ecf/defs/prod12.def:
```diff
@@ -3,30 +3,38 @@ extern /prod06/gdas/atmos/post
+extern /prod18/gdas/atmos/analysis/jgdas_atmos_analysis
+extern /prod18/gfs/atmos/analysis/jgfs_atmos_analysis
+extern /prod18/gdas/enkf/analysis/recenter/ecen
+extern /totality_limit:TOTALITY
```
Is this not necessary in the other `prod00.def`, `prod06.def` and `prod18.def` files?

aerorahul commented on ecf/defs/prod00.def:
```diff
@@ -3,30 +3,38 @@ extern /prod18/gdas/atmos/post
 extern /prod18/gdas/enkf/post
 #### ecen
 extern /prod18/gdas/atmos/post_processing/jgdas_atmos_chgres_forenkf
+extern /prod06/gdas/atmos/analysis/jgdas_atmos_analysis
+extern /prod06/gfs/atmos/analysis/jgfs_atmos_analysis
+extern /prod06/gdas/enkf/analysis/recenter/ecen
```
I am curious what in the 00z suite is depending on a trigger from 06z.
I don't see any references to triggers from these.

As a matter of fact, the only triggers are from 
`/prod18/gdas/enkf/post` and `/prod18/gdas/atmos/post_processing/jgdas_atmos_chgres_forenkf`

WalterKolczynski-NOAA replied:
I had the same question. Why are new triggers being added when this is only supposed to be a port? And why triggers from three cycles previous?

WalterKolczynski-NOAA commented:
There are a bunch of places where ops settings have been replaced by dev ones. I know they are needed for us to run it, but we are very close to hand-off, so I don't know whether I should be pointing them all out or not.
